deneb.go

```diff
@@ -0,0 +1,224 @@
+			Index:          uint64(withdrawal.Index),
+			Amount:         uint64(withdrawal.Amount),
+			Address:        withdrawal.Address[:],
+		})
+	}
+
+	return &SubmitBlockRequest{
+		BidTrace: &BidTrace{
+			Slot:                 block.Message.Slot,
+			ParentHash:           block.Message.ParentHash[:],
+			BlockHash:            block.Message.BlockHash[:],
+			BuilderPubkey:        block.Message.BuilderPubkey[:],
+			ProposerPubkey:       block.Message.ProposerPubkey[:],
+			ProposerFeeRecipient: block.Message.ProposerFeeRecipient[:],
+			GasLimit:             block.Message.GasLimit,
+			GasUsed:              block.Message.GasUsed,
+			Value:                block.Message.Value.Hex(),
+			BlobGasUsed:          block.ExecutionPayload.BlobGasUsed,
+			ExcessBlobGas:        block.ExecutionPayload.ExcessBlobGas,
+		},
+		ExecutionPayload: &ExecutionPayload{
+			ParentHash:    block.ExecutionPayload.ParentHash[:],
+			StateRoot:     block.ExecutionPayload.StateRoot[:],
+			ReceiptsRoot:  block.ExecutionPayload.ReceiptsRoot[:],
+			LogsBloom:     block.ExecutionPayload.LogsBloom[:],
+			PrevRandao:    block.ExecutionPayload.PrevRandao[:],
+			BaseFeePerGas: uint256ToIntToByteSlice(block.ExecutionPayload.BaseFeePerGas),
+			FeeRecipient:  block.ExecutionPayload.FeeRecipient[:],
+			BlockHash:     block.ExecutionPayload.BlockHash[:],
+			ExtraData:     block.ExecutionPayload.ExtraData,
+			BlockNumber:   block.ExecutionPayload.BlockNumber,
+			GasLimit:      block.ExecutionPayload.GasLimit,
+			Timestamp:     block.ExecutionPayload.Timestamp,
+			GasUsed:       block.ExecutionPayload.GasUsed,
+			Transactions:  compressTxs,
+			Withdrawals:   withdrawals,
+			BlobGasUsed:   block.ExecutionPayload.BlobGasUsed,
+			ExcessBlobGas: block.ExecutionPayload.ExcessBlobGas,
+		},
+		BlobsBundle: convertBlobBundleToProto(block.BlobsBundle),
+		Signature:   block.Signature[:],
+	}
+}
+
+func ProtoRequestToDenebRequest(block *SubmitBlockRequest) *apiDeneb.SubmitBlockRequest {
+	transactions := []bellatrix.Transaction{}
+	for _, tx := range block.ExecutionPayload.Transactions {
+		transactions = append(transactions, tx.RawData)
+	}
+
+	// Withdrawal is defined in capella spec
+	// https://github.com/attestantio/go-eth2-client/blob/21f7dd480fed933d8e0b1c88cee67da721c80eb2/spec/deneb/executionpayload.go#L42
+	withdrawals := []*capella.Withdrawal{}
+	for _, withdrawal := range block.ExecutionPayload.Withdrawals {
+		withdrawals = append(withdrawals, &capella.Withdrawal{
+			ValidatorIndex: phase0.ValidatorIndex(withdrawal.ValidatorIndex),
+			Index:          capella.WithdrawalIndex(withdrawal.Index),
+			Amount:         phase0.Gwei(withdrawal.Amount),
+			Address:        b20(withdrawal.Address),
+		})
+	}
+
+	// BlobsBundle
+	blobsBundle := &apiDeneb.BlobsBundle{}
+	for _, commitment := range block.BlobsBundle.Commitments {
+		// Unmarshal commitment
+		var commitmentObj consensus.KZGCommitment
+		err := commitmentObj.UnmarshalJSON(commitment)
```

Comment:
should copy bytes for these too since `marshalJSON()` was removed in the reverse function

Reply:
done